dm/pkg/schema/tracker.go

```diff
@@ -233,22 +242,40 @@ func NewTracker(
 	// exist by default. So we need to drop it first.
 	err = dom.DDL().DropSchema(se, model.NewCIStr("test"))
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	// init downstreamTracker
 	dsTracker := &downstreamTracker{
 		downstreamConn: downstreamConn,
 		tableInfos:     make(map[string]*DownstreamTableInfo),
 	}
+	tr.Lock()
+	defer tr.Unlock()
```

Comment:
now this lock is used in 3 places, please refine comments in `Tracker` definition

Reply:
[77cfbca](https://github.com/pingcap/tiflow/pull/6052/commits/77cfbcab957523dfd1092a03463b27183f28436f)